main.go

```diff
@@ -190,7 +190,7 @@ func main() {
 	}
 
 	// Initialize and monitor shutdown signal
-	shutdownSignal = make(chan bool, 1)
+	prepareGracefulShutdown()
 	monitorShutdownSignal(os.Exit)
```

Comment:
This function can go potentially inside 'initGracefulShutdown()' - that way everywhere only one function has to be called.